---
title: "SRP096177"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document
---
Source: https://github.com/markziemann/gene_sig_commons

## Background

This analysis is an example of how gene signatures can be generated from publicly available omics data.
In this example, we will obtain data from the [DEE2](http://dee2.io) database, perform differential analysis with DESeq2 and then return the gene signature. 

```{r,setup}
suppressPackageStartupMessages({
    library("devtools")
    # devtools::install_github("markziemann/dee2/getDEE2") # Version: 0.0.4
    library("getDEE2")
    library("DESeq2")
    library("edgeR")
    library("gplots") 
})
```



```{r,study_list}
# let's assume species is human for now
SPECIES = "hsapiens"

# read in the file
x <- readLines("epilepsy.md")

# this is the best way to get studies with contrasts set up
x <- x[grep("RP",x)]

# lets prototype this by 
x1 <- x[[1]]

# define the SRP
SRP = strsplit(x1,":")[[1]][1]

# define contrast name
contrast_name = strsplit(x1,":")[[1]][2]

# get control group info
control = strsplit(x1,":")[[1]][3]

# get control group name
control_name = strsplit(control,";")[[1]][1]

# get control groups samples
# note that we are extracting SRX data not SRR
control_data = strsplit(control,";")[[1]][2]
control_data <- unlist(strsplit(control_data, ","))
control_data <- gsub(" ","",control_data)

# get case group info
case = strsplit(x1,":")[[1]][4]

# get case group name
case_name = strsplit(case,";")[[1]][1]

# get case groups samples
case_data = strsplit(case,";")[[1]][2]
case_data <- unlist(strsplit(case_data, ","))
case_data <- gsub(" ","",case_data)

# set up the sample sheet starting with a list of samples
ss <- as.data.frame(paste(c(control_data,case_data),sep=","))

# define case group
ss$case <- as.numeric(ss[,1] %in% case_data)

# set up the model
mm <- model.matrix(~case,ss)
rownames(mm) <- ss[,1]
mm
```
## Download

Fetch the dataset from [DEE2](http://dee2.io).

```{r,get}
mdat <- getDee2Metadata(species = SPECIES)

SRRvec <- as.character(mdat[which(mdat$SRP_accession==SRP),1])
SRRvec
x <- getDEE2(SRRvec = SRRvec, species = SPECIES, metadata = mdat)
str(x)
head(x$GeneCounts)

```

## Multidimensional scaling plot

MDS analysis is a bit like principal component analysis. 
The closer the labels are, the more similar the datasets are.

```{r,analysis1}
y <- x$GeneCounts
colnames(y) <- samplesheet$label
# Remove genes with fewer than 10 reads per sample on average
yy <- y[which(rowMeans(y)>10),]
MDS <- function(x,mylabels, ...) {
    mydist <- cmdscale(dist(t(x)))
    myrange <- range(mydist[,1])*1.3
    plot(mydist, xlab="Coordinate 1", ylab="Coordinate 2", 
        type = "n", xlim=myrange, ...)
    text(mydist, labels=mylabels, cex=0.9) 
}
MDS(x = yy, mylabels = samplesheet$label ,main=SRP)

```

## Differential analysis

```{r,de}
dds <- DESeqDataSetFromMatrix(countData = yy,
    colData = samplesheet, design= mm )
dds <- DESeq(dds)
de <- results(dds)

# RPM
yyy <- yy/colMeans(yy) * 1000000
res <- cbind(de,yyy,yy)

res <- res[order(res$pvalue),]
head(res)

# define up and down-regulated gene lists
up <- rownames(subset(de, log2FoldChange>0 & padj<0.05 ))
dn <- rownames(subset(de, log2FoldChange<0 & padj<0.05 ))
```

## Charts

```{r,decharts}
# MA plot
sig <-subset(de, padj < 0.05 )
GENESUP <- length(up)
GENESDN <- length(dn)
SUBHEADER = paste(GENESUP, "up, ", GENESDN, "down")
ns <-subset(de, padj > 0.05 )
plot(log2(de$baseMean),de$log2FoldChange, 
     xlab="log2 basemean", ylab="log2 foldchange",
     pch=19, cex=0.5, col="dark gray",
     main=CONTRASTNAME)
points(log2(sig$baseMean),sig$log2FoldChange,
       pch=19, cex=0.5, col="red")
mtext(SUBHEADER)

# heatmap
top <- res[1:50,7:ncol(res)]
top <- top[,1:(ncol(top)/2)]
colfunc <- colorRampPalette(c("blue", "white", "red"))
heatmap.2(  as.matrix(top), col=colfunc(25),scale="row", trace="none",
    margins = c(6,6), cexRow=.4, main=SRP)
```

## Hiearchical clustering

```{r,clustering}
library(RColorBrewer)
# Generate some random data
N_SAMPLES=20
N_GENES=30
x<- matrix(data = rexp(600, rate = 10), nrow = N_GENES, ncol = N_SAMPLES)
rownames(x) <- paste("genes",1:N_GENES)
colnames(x) <- paste("sample",1:N_SAMPLES)
head(x)

cl<-as.dist(1-cor(t(x), method="spearman"))
hr <- hclust(cl , method="complete")
mycl <- cutree(hr, h=max(hr$height/1.2))
clusterCols <- brewer.pal(length(unique(mycl)),"Paired")
myClusterSideBar <- clusterCols[mycl]
colfunc <- colorRampPalette(c("blue", "white", "red"))
write.table(mycl,file="GeneClusters1.txt",quote=F,sep="\t")

# create heatmap
heatmap.2(x, main="Gene Clustering 1",  Rowv=as.dendrogram(hr),
 dendrogram="both", scale="none", col = colfunc(25), trace="none", 
 RowSideColors= myClusterSideBar, margins = c(5,5))

```

##  Upload
```{r,genesets}
DENAME = paste(SRP, "differential expression by knock-down of Setd7 expression")
UPNAME = paste(SRP, "up-regulated by knock-down of Setd7 expression")
DOWNNAME = paste(SRP, "down-regulated by knock-down of Setd7 expression")
df <- data.frame(de)
gs <- list(df,up,dn)
names(gs) <- c(DENAME,UPNAME,DOWNNAME)
str(gs)
# TODO: secure copy
# https://www.rdocumentation.org/packages/ssh/versions/0.6/topics/scp
```

## Session info
```{r,sessioninfo}
sessionInfo()
```