---
title: "SRP096177"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document
---
Source: https://github.com/markziemann/gene_sig_commons

## Background

This analysis is an example of how gene signatures can be generated from publicly available omics data.
In this example, we will obtain data from the [DEE2](http://dee2.io) database, perform differential analysis with DESeq2 and then return the gene signature. 

```{r,setup}
suppressPackageStartupMessages({
    library("devtools")
    # devtools::install_github("markziemann/dee2/getDEE2") # Version: 0.0.4
    library("getDEE2")
    library("DESeq2")
    library("edgeR")
    library("gplots") 
})
```

## Study Description

SRA project ID: SRP096177

GEO Series ID: GSE93236

Project description: https://www.ncbi.nlm.nih.gov//geo/query/acc.cgi?acc=GSE93236

Publication: https://pubmed.ncbi.nlm.nih.gov/24875254/

In this study, the data contributors aimed to ascertain the genes controlled by the protein Setd7. 
The two groups are control (ctrl, n=3) and knock-down (Set7 KD,n=3). 
In the knock-down samples, SETD7 expression was reduced by 90%.

```{r,study_info}
SRP = "SRP096177"
SPECIES = "hsapiens"
RUNS <- c("SRR5150592","SRR5150593","SRR5150594","SRR5150595","SRR5150596","SRR5150597")
GROUPS <- factor(c("Set7KD","Set7KD","Set7KD","NTC","NTC","NTC"), 
    levels=c("NTC","Set7KD"),ordered = TRUE)
CONTRASTNAME="Control vs Set7 knock-down"
samplesheet <- as.data.frame(GROUPS,row.names = RUNS)
samplesheet$label <- paste(rownames(samplesheet),samplesheet$GROUPS)
mm <- model.matrix(~GROUPS,samplesheet)
mm
```
## Download

Fetch the dataset from [DEE2](http://dee2.io).

```{r,get}
mdat <- getDee2Metadata(species = SPECIES)
SRRvec <- as.character(mdat[which(mdat$SRP_accession==SRP),1])
SRRvec
x <- getDEE2(SRRvec = SRRvec, species = SPECIES, metadata = mdat)
str(x)
head(x$GeneCounts)

```

## Multidimensional scaling plot

MDS analysis is a bit like principal component analysis. 
The closer the labels are, the more similar the datasets are.

```{r,analysis1}
y <- x$GeneCounts
colnames(y) <- samplesheet$label
# Remove genes with fewer than 10 reads per sample on average
yy <- y[which(rowMeans(y)>10),]
MDS <- function(x,mylabels, ...) {
    mydist <- cmdscale(dist(t(x)))
    myrange <- range(mydist[,1])*1.3
    plot(mydist, xlab="Coordinate 1", ylab="Coordinate 2", 
        type = "n", xlim=myrange, ...)
    text(mydist, labels=mylabels, cex=0.9) 
}
MDS(x = yy, mylabels = samplesheet$label ,main=SRP)

```

## Differential analysis

```{r,de}
dds <- DESeqDataSetFromMatrix(countData = yy,
    colData = samplesheet, design= mm )
dds <- DESeq(dds)
de <- results(dds)

# RPM
yyy <- yy/colMeans(yy) * 1000000
res <- cbind(de,yyy,yy)

res <- res[order(res$pvalue),]
head(res)

# define up and down-regulated gene lists
up <- rownames(subset(de, log2FoldChange>0 & padj<0.05 ))
dn <- rownames(subset(de, log2FoldChange<0 & padj<0.05 ))
```

## Charts

```{r,decharts}
# MA plot
sig <-subset(de, padj < 0.05 )
GENESUP <- length(up)
GENESDN <- length(dn)
SUBHEADER = paste(GENESUP, "up, ", GENESDN, "down")
ns <-subset(de, padj > 0.05 )
plot(log2(de$baseMean),de$log2FoldChange, 
     xlab="log2 basemean", ylab="log2 foldchange",
     pch=19, cex=0.5, col="dark gray",
     main=CONTRASTNAME)
points(log2(sig$baseMean),sig$log2FoldChange,
       pch=19, cex=0.5, col="red")
mtext(SUBHEADER)

# heatmap
top <- res[1:50,7:ncol(res)]
top <- top[,1:(ncol(top)/2)]
colfunc <- colorRampPalette(c("blue", "white", "red"))
heatmap.2(  as.matrix(top), col=colfunc(25),scale="row", trace="none",
    margins = c(6,6), cexRow=.4, main=SRP)
```

##  Upload
```{r,genesets}
DENAME = paste(SRP, "differential expression by knock-down of Setd7 expression")
UPNAME = paste(SRP, "up-regulated by knock-down of Setd7 expression")
DOWNNAME = paste(SRP, "down-regulated by knock-down of Setd7 expression")
df <- data.frame(de)
gs <- list(df,up,dn)
names(gs) <- c(DENAME,UPNAME,DOWNNAME)
str(gs)
# TODO: secure copy
# https://www.rdocumentation.org/packages/ssh/versions/0.6/topics/scp
```

## Session info
```{r,sessioninfo}
sessionInfo()
```